### PR TITLE
drivers: axi_core: axi_sysid: Add driver

### PR DESCRIPTION
**Pull request by danmois**
## Pull Request Description

Add the driver for the System ID core.

Required for detecting carrier board for Xilinx projects. This will simplify parameters.h.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

File: drivers/axi_core/axi_sysid/axi_sysid.c [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+	struct sysid_header_v1 *header;
+	char custom_info[516];
+
+	if (axi_sysid_checksum((uint8_t *)build,
+			       sizeof(struct build_info_header_v1_1))) {
+		pr_err("verify build header checksum failed\n");
+		return -EFAULT;
+	}
+
+	header = (struct sysid_header_v1 *) st->mem;
```
Comment (amiclaus) [inline drivers/axi_core/axi_sysid/axi_sysid.c:123]: nit: no need to add space after type cast.

File: drivers/axi_core/axi_sysid/axi_sysid.c [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+	info = (struct axi_sysid_core_info *)no_os_calloc(1, sizeof(*info));
+	if (!info)
+		goto error;
+
+	info->version = axi_sysid_ioread(sysid, AXI_SYSID_REG_VERSION);
+
+	sysid->info = info;
+
+	switch(sysid->info->version) {
+	case 0x10061:
```
Comment (amiclaus) [inline drivers/axi_core/axi_sysid/axi_sysid.c:244]: some macros would be nice.

File: drivers/axi_core/axi_sysid/axi_sysid.c [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+		goto error1;
+
+	*sysid_core = sysid;
+
+	return 0;
+error1:
+	no_os_free(info);
+error:
+	no_os_free(sysid);
+	return -1;
```
Comment (amiclaus) [inline drivers/axi_core/axi_sysid/axi_sysid.c:263]: hmm, can we use some specific error code here?

File: drivers/axi_core/axi_sysid/axi_sysid.c [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+* @brief Remove the SYSID instance.
+*
+* @param sysid - SYSID istance.
+*
+* @return 0 for success, -1 in case of failure.
+*******************************************************************************/
+int32_t axi_sysid_remove(struct axi_sysid *sysid)
+{
+	if (!sysid)
+		return -1;
```
Comment (amiclaus) [inline drivers/axi_core/axi_sysid/axi_sysid.c:276]: ditto.

**Review by amiclaus — COMMENTED** [review]
looks good overall! minor comments.

**Comment by danmois** [comment]
V2
- use of specific error codes in case of error
- use macros for version
- remove unnecessary space

File: drivers/axi_core/axi_sysid/axi_sysid.c (hunk 4 above)
Comment (danmois) [reply drivers/axi_core/axi_sysid/axi_sysid.c:281]: Thank you for the comments. The code has been updated accordingly.

**Review by amiclaus — APPROVED** [review]

**Pull request merged**